magenta/interfaces/midi/midi_hub.py

```diff
@@ -177,22 +192,40 @@ def run(self):
         self._start_time,
         now + period - ((now - self._start_time) % period))
     while self._stop_time is None or self._stop_time > next_tick_time:
+      self._count += 1
       sleeper.sleep_until(next_tick_time)
 
-      self._outport.send(
+      if self._count % self._qpb == 1:
```

Comment:
use the conditional just to choose the pitch and velocity on each iteration instead of duplicating the send/sleep code